### DISCUSSION TITLE
Paper not aligned (0,0) with paperScroller in a Autoresize:true scenario

### DISCUSSION THREAD

**Discussion by jibinjoy8897** [clientIO/joint#2526] [Q&A] [answered]
### Introduction

const scroller = new ui.PaperScroller({
      paper,
      autoResizePaper: true,
    });
canvasEl.append(scroller.render().el);

These are my changes for adding paperScroller on paper.But my paper is not aligned to (0,0) of paperScroller there is a lot of space around the paper.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Duplicate of https://github.com/clientIO/joint/discussions/2182?

**Reply by jibinjoy8897** [reply]
Duplicate of #2344 .Added borderless:true.Note:(directedLayout is used) .
Issue:
**g tag inside svg is offset from the left and top**

**Reply by kumilingus** [reply]
To be honest I didn't understand https://github.com/clientIO/joint/issues/2344 back then and I don't understand now.

> tag inside svg is offset from the left and top

Why is this a problem?

**Reply by jibinjoy8897** [reply]
The customer does not want the nodes to go outside to the negative coordinates of the paper.Added below code to do that

 // Restrict position of element to go to negative coordinates in canvas

    paper.on({
      'element:pointermove': function (elementView, evt, x, y) {
        let element = elementView.model;
        if (element.changed?.position) {
          let changedXPos = element.changed.position.x;
          let changedYPos = element.changed.position.y;
          element.position(changedXPos < 0 ? 0 : changedXPos, changedYPos < 0 ? 0 : changedYPos);
        }
      },
    });
After this change noticed that the left and top most edge of the g tag is considered as origin of diagram (0,0) and there is a extra space between the svg and the child g tag,which makes it look odd,because we expect the Origin to start from the top most and left most of a div.

**Reply by kumilingus** [reply]
1. If you don't want the paper to expand to **top / left** use `contentOptions`:
   ```js
   new joint.ui.PaperScroller({
        paper,
        autoResizePaper: true,
        contentOptions: {
            allowNewOrigin: false,
        }
    });
    ```
2. To restrict translate of an element use [restrictTranslate](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Paper.prototype.options.restrictTranslate) paper option.

**Comment by jibinjoy8897** [comment]
> 1. If you don't want the paper to expand to **top / left** use `contentOptions`:
>    ```js
>    new joint.ui.PaperScroller({
>         paper,
>         autoResizePaper: true,
>         contentOptions: {
>             allowNewOrigin: false,
>         }
>     });
>    ```
> 2. To restrict translate of an element use [restrictTranslate](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Paper.prototype.options.restrictTranslate) paper option.

**Reply by jibinjoy8897** [reply]
1) Did u mean allowNewOrigin: positive, as  allowNewOrigin: false, is throwing error.
2) restrictTranslate worked perfectly.Thanks

**Reply by kumilingus** [reply]
No, I meant the default for `allowNewOrigin` which is `"do not allow new origin"` (But the *TypeScript* does not like it).
Does `allowNewOrigin: null` work?
If not, please use `@ts-ignore` for the time being. It needs to be updated in the type definition.

**Reply by jibinjoy8897** [reply, marked as answer]
1) allowNewOrigin in documentation says its default is positive.Other options are 'negative' or 'any'.I dont understand why you suggested (allowNewOrigin: false,)

Changes made to make it work:
paperScroller settings ->(borderless:true)
paperScroller settings ->(restrictTranslate: true,)
This solved my problems.
Thanks for the quick response

**Reply by kumilingus** [reply]
Please see the following PR that explains `allowNewOrigin: false`: https://github.com/clientIO/joint/pull/2530
